### ISSUE TITLE
ejabberd 17.07 ejabberdctl scripts execution errors

### ISSUE THREAD

**Issue by cmeng-git** [processone/ejabberd#1933]
> What version of ejabberd are you using?
ejabberd 17.07+

> What operating system (version) are you using?
ubuntu 16.04

> How did you install ejabberd (source, package, distribution)?

source build from master trunk using the following configuration settings:
`sudo ./configure --prefix=/ --exec_prefix=/usr --datadir=/usr/share --enable-pam --enable-mysql --enable-iconv --enable-stun --enable-sip --enable-latest-deps --enable-user=ejabberd`

//---------------------------------//
Followings are the system environment setup
Both the ejabberd and ejabberdctl are installed from latest source 17.07+
System is running Ubuntu 16.04
```
erl --version gives
Erlang/OTP 20 [erts-9.0] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false]

```
Also I have to change the following variable in sbin.ejabberd to work with the latest erl version.
POLL=true
SMP=auto
//-------------------------------//

> What did not work as expected? Are there error messages in the log? What
I am facing problem in starting up ejabberd with the latest scripts: /etc/init.d/ejabberd and /usr/sbin/ejabberdctl 

None of the following commands work i.e. sudo ejabberdctl start|stop|live|status. Neither can I start ejabberd using sudo /etc/init.d/ejabberd start or on Ubuntu system start up.

//---------------------------------//
When I use sudo ejabberdctl start | live to start, I receive the following error messages on the terminal:
`erl_child_setup closed
`
and the only process get started is
/usr/lib/erlang/erts-9.0/bin/epmd -daemon

//--------------------------------//
If I start ejabberd using sudo /etc/init.d/ejabberd start, then I received the following error messages:
Starting ejabberd...
```
erl_child_setup closed

Crash dump is being written to: erl_crash.dump...done
erl_child_setup closed

Crash dump is being written to: erl_crash.dump...done
erl_child_setup closed

Crash dump is being written to: //var/log/ejabberd/erl_crash_20170805-085700.dump...done
erl_child_setup closed

Crash dump is being written to: //var/log/ejabberd/erl_crash_20170805-085700.dump...done
erl_child_setup closed

Crash dump is being written to: //var/log/ejabberd/erl_crash_20170805-085700.dump...done

```
like before, the only process get started is
/usr/lib/erlang/erts-9.0/bin/epmd -daemon

//------------------//
The only thing works right now is a small script that I added in /etc/init.d/ejabberd and
execute with sudo /etc/init.d/ejabberd live

```
//============= live script in /etc/init.d/ejabberd ==========/

EJABBERD="$DIR"/ejabberd
  live)
    test -x "$CTL" || exit 0
    echo "Stopping ejabberd live ..."
    echo '*******************************************************'
    echo '* To quit, press Ctrl-g then enter q and press Return *'
    echo '*******************************************************'
    echo

    cd /var/lib/ejabberd
    su - $USER -c "$EJABBERD"
    echo "done."
    ;;
//=================//
```

Another observation is that
/run/ejabberd/ejabber.pid; both the pid file and /run/ejabberd file path get deleted on system reboot. So every time I reboot the system, I have to manually re-create the path /run/ejabberd as follow before I can run sudo /etc/init.d/ejabberd live.
`drwxr-xr-x 2 ejabberd root 80 Jul 31 11:14 ejabberd`

If I just start with command using /usr/sbin/ejabberd script i.e. not via ejabberdctl script generation
sudo ejabberd  OR
sudo ejabberd -noinput -detached
then ejabberd starts and works as expected; the actual shell script executed is as below

```
$ sudo ejabberd -noinput -detached
/usr/bin/erl -sname ejabberd -mnesia dir "/var/lib/ejabberd" -kernel inetrc "/etc/ejabberd/inetrc" -ejabberd config "/etc/ejabberd/ejabberd.yml" log_path "/var/log/ejabberd/ejabberd.log" -s ejabberd -pa /usr/lib/ejabberd/ebin erlang_log_path "/var/log/ejabberd/erlang.log" -sasl sasl_error_logger false +K true -smp auto +P 250000 -noinput -detached

```
> What was the unexpected behavior? What was the expected result?
See above

It seems like the problem lies within ejabberdctl script generation.
Any help is much appreciated.

**Comment by cromain** [comment]
installing ejabberd as root and starting it as root works out of the box with HEAD, just 'make install' and 'ejabberdctl start'
why you configure compile install and run as root is a big mystery BTW.
your error is likely due to your environment and/or OS settings.

**Comment by cmeng-git** [comment]
My initial ejabberd installation is from Ubuntu distributed package (since 2.17); its installed directories is slightly different from the default ejabberd source built installation. Since then, to be compatible with Ubuntu distributed package installed directories, I always perform ./configure before any new built; and to enable certain package which are not enabled by default. I also found that if there is any new dep package added, you must perform configure, otherwise the deps lib will not be updated properly.

When building from source, I always follow the steps below:
•	$ sudo ./configure --prefix=/ --exec_prefix=/usr --datadir=/usr/share --enable-pam --enable-mysql --enable-iconv --enable-stun --enable-sip --enable-latest-deps --enable-user=ejabberd
•	$ make clean
•	$ ./rebar get-deps  (ensure that chmod +x ./rebar has been performed before using rebar command)
•	$ make update (update deps sources for compilation – if there are new added deps)
•	$ make all –jx  (make all including deps and and x is the number of cpu processors/core)
•	$ sudo checkinstall (to install and generate deb package)

Instead of using make install, I use checkinstall. The checkinstall tool ensures that the previous installed directories are properly cleanup before new installation. it also create the .deb package for future reference.

Actually I did try out to remove --enable-user=ejabberd, but it did not help.
https://github.com/processone/ejabberd/issues/1561#event-1193047430

Since I am able to start the ejabberd with the resulted command being executed as shown below
$ sudo ejabberd -noinput -detached

/usr/bin/erl -sname ejabberd -mnesia dir "/var/lib/ejabberd" -kernel inetrc "/etc/ejabberd/inetrc" -ejabberd config "/etc/ejabberd/ejabberd.yml" log_path "/var/log/ejabberd/ejabberd.log" -s ejabberd -pa /usr/lib/ejabberd/ebin erlang_log_path "/var/log/ejabberd/erlang.log" -sasl sasl_error_logger false +K true -smp auto +P 250000 -noinput -detached

This leads me to believe that the system environment for erl execution is setup correctly.
Any execution via ejabberdctl immediately causes erl to exit
erl_child_setup closed

This is the reason why I came to the conclusion that ejabberdctl script generation has included some options not like by erl.
